tests/check_discovery.c

```diff
@@ -411,9 +411,10 @@ static void GetEndpointsAndCheck(const char* discoveryUrl, const char* filterTra
 
     ck_assert_uint_eq(endpointArraySize , expectedEndpointUrlsSize);
 
-    for(size_t j = 0; j < endpointArraySize; j++) {
+    for(size_t j = 0; j < endpointArraySize && j < expectedEndpointUrlsSize; j++) {
         UA_EndpointDescription* endpoint = &endpointArray[j];
         char *eu = UA_String_to_char_alloc(&endpoint->endpointUrl);
+        ck_assert_ptr_ne(eu, NULL); // clang static analysis fix
```

Comment:
Hey @Pro ,

embedded applications are often on a tight budget ram-wise. We can never know when memory runs out.
So the rest of the SDK always checks whether malloc succeeds.

The assertion patches up the clang-analyzer issue.
But it will still segfault the moment malloc actually returns NULL.

Once clang-analyzer is happy with the code (and you are too), we can push the branch to coverity.
Coverity will probably uncover a couple more issues. That's normal for a feature this big. 😄